apps/nestjs-wallet/src/vc-api/exchanges/entities/transaction.entity.ts

```diff
@@ -85,8 +134,16 @@ export class TransactionEntity {
     response: ExchangeResponseDto;
     callback: CallbackConfiguration[];
   } {
-    // TODO check that submitted presentation matches the vpRequest
-    // Checks could be similar to https://github.com/gataca-io/vui-core/blob/46352ccff298eb1d237e4072d982768d79001041/service/validatorServiceDIFPE.go#L54
+    try {
+      this.validatePresentation(presentation);
+    } catch (error) {
```

Comment:
IMO, it's not a good idea generally speaking to use exceptions for control flow (https://softwareengineering.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why)
I think it would better to return an `errors` array if validation fails (as is done elsewhere in the code thus far)

Reply:
Yeah, you're right